apiserver/facades/client/applicationoffers/base.go

```diff
@@ -53,31 +55,27 @@ func (api *BaseAPI) checkAdmin(
 
 // checkControllerAdmin ensures that the logged in user is a controller admin.
 func (api *BaseAPI) checkControllerAdmin(ctx context.Context) error {
-	return api.Authorizer.HasPermission(ctx, permission.SuperuserAccess, api.ControllerModel.ControllerTag())
+	controllerTag := names.NewControllerTag(api.controllerUUID)
+	return api.Authorizer.HasPermission(ctx, permission.SuperuserAccess, controllerTag)
 }
 
-// modelForName looks up the model details for the named model and returns
-// the model (if found), the absolute model path which was used in the lookup,
-// and a bool indicating if the model was found,
-func (api *BaseAPI) modelForName(modelName, ownerName string) (Model, string, bool, error) {
+// modelForName returns the model details for the specified model name,
+// along with the absolute model path used in the lookup.
+//
+// The following errors may be returned:
+// - [modelerrors.NotFound] when no model with the given name exists.
+// - [coreerrors.NotValid] when ownerName is not a valid Name object.
+func (api *BaseAPI) modelForName(ctx context.Context, modelName, ownerName string) (*model.Model, string, error) {
 	modelPath := fmt.Sprintf("%s/%s", ownerName, modelName)
-	var model Model
-	uuids, err := api.ControllerModel.AllModelUUIDs()
+	ownerUsername, err := coreuser.NewName(ownerName)
 	if err != nil {
-		return nil, modelPath, false, errors.Trace(err)
+		return nil, modelPath, errors.Trace(err)
 	}
-	for _, uuid := range uuids {
-		m, release, err := api.StatePool.GetModel(uuid)
-		if err != nil {
-			return nil, modelPath, false, errors.Trace(err)
-		}
-		defer release()
-		if m.Name() == modelName && m.Owner().Id() == ownerName {
-			model = m
-			break
-		}
+	model, err := api.modelService.GetModelByNameAndOwner(ctx, modelName, ownerUsername)
+	if errors.Is(err, modelerrors.NotFound) {
```

Comment:
I think you want to let this error return out of this func. The caller has a specific nil check to detect this case when it could just be using this value and the error check that is much more reliable.

Reply:
Sure thanks tom